benchmark/utils/overhead_linop.hpp

```diff
@@ -0,0 +1,196 @@
+#ifndef GKO_BENCHMARK_UTILS_OVERHEAD_LINOP_
+#define GKO_BENCHMARK_UTILS_OVERHEAD_LINOP_
+
+
+#include <cstdint>
+#include <memory>
+#include <vector>
+
+
+#include <ginkgo/core/base/lin_op.hpp>
+#include <ginkgo/core/base/types.hpp>
+#include <ginkgo/core/stop/criterion.hpp>
+
+
+namespace gko {
+namespace kernels {
+namespace overhead {
+
+
+#define GKO_DECLARE_OVERHEAD_OPERATION_KERNEL(_type, _num)            \
+    static volatile std::uintptr_t val_operation_##_num = 0;          \
+    template <typename _type>                                         \
+    void operation##_num(std::shared_ptr<const DefaultExecutor> exec, \
+                         const matrix::Dense<_type> *b,               \
+                         matrix::Dense<_type> *x)                     \
+    {                                                                 \
+        val_operation_##_num = reinterpret_cast<std::uintptr_t>(x);   \
+    }
+
+
+#define GKO_DECLARE_ALL                                                      \
+    GKO_DECLARE_OVERHEAD_OPERATION_KERNEL(ValueType, 1)                      \
+    GKO_DECLARE_OVERHEAD_OPERATION_KERNEL(ValueType, 2)                      \
+    GKO_DECLARE_OVERHEAD_OPERATION_KERNEL(ValueType, 3)                      \
+    GKO_DECLARE_OVERHEAD_OPERATION_KERNEL(ValueType, 4)                      \
+    static_assert(true,                                                      \
+                  "This assert is used to counter the false positive extra " \
+                  "semi-colon warnings")
+
+
+}  // namespace overhead
+
+
+namespace omp {
+namespace overhead {
+
+GKO_DECLARE_ALL;
+
+}  // namespace overhead
+}  // namespace omp
+
+
+namespace cuda {
+namespace overhead {
+
+GKO_DECLARE_ALL;
+
+}  // namespace overhead
+}  // namespace cuda
+
+
+namespace reference {
+namespace overhead {
+
+GKO_DECLARE_ALL;
+
+}  // namespace overhead
+}  // namespace reference
+
+
+namespace hip {
+namespace overhead {
+
+GKO_DECLARE_ALL;
+
+}  // namespace overhead
+}  // namespace hip
+
+
+#undef GKO_DECLARE_ALL
+
+
+}  // namespace kernels
+
+
+namespace overhead {
+
+
+GKO_REGISTER_OPERATION(operation1, overhead::operation1);
+GKO_REGISTER_OPERATION(operation2, overhead::operation2);
+GKO_REGISTER_OPERATION(operation3, overhead::operation3);
+GKO_REGISTER_OPERATION(operation4, overhead::operation4);
+
+
+}  // namespace overhead
+
+
+template <typename ValueType = default_precision>
+class Overhead : public EnableLinOp<Overhead<ValueType>>,
+                 public Preconditionable {
+    friend class EnableLinOp<Overhead>;
+    friend class EnablePolymorphicObject<Overhead, LinOp>;
+
+public:
+    using value_type = ValueType;
```

Comment:
```suggestion
```
`value_type` is unused here.

Reply:
I removed it here since that is not something exposed to users, but as a side note these are usually unused in the code's class itself. They can be used from outside the class.

Reply:
You are right. I forgot that it can be used from outside.
I am okay for keeping or removing it.